src/mqtt/MQTT.cpp

```diff
@@ -482,7 +482,8 @@ void MQTT::onSend(const meshtastic_MeshPacket &mp, const meshtastic_MeshPacket &
 
     auto &ch = channels.getByIndex(chIndex);
 
-    if (&mp_decoded.decoded && strcmp(moduleConfig.mqtt.address, default_mqtt_address) == 0 &&
+    assert(mp_decoded.which_payload_variant == meshtastic_MeshPacket_decoded_tag);
```

Comment:
Please don't introduce new asserts. We're trying to get rid of them.

Reply:
Done.